ssl/record/ssl3_buffer.c

```diff
@@ -107,6 +107,11 @@ int ssl3_setup_write_buffer(SSL *s, size_t numwpipes, size_t len)
     for (currpipe = 0; currpipe < numwpipes; currpipe++) {
         SSL3_BUFFER *thiswb = &wb[currpipe];
 
+        if (thiswb->buf != NULL && thiswb->len < len) {
```

Comment:
Also, we still only seem to be doing the write buffer. What about the read buffer?